File: queues/rabbit/C/tests/test_rabbitclient.c
#include "../../../../tests/c/testutil.h"
#include "../include/rabbitclient.h"
#include <stdlib.h>

TEST(test_init_missing)
{
    unsetenv_safe("RABBITMQ_URL");
    unsetenv_safe("RABBITMQ_HOST");
    char *json = rabbit_init();
    ASSERT("601", json_status(json) == 601);
    free(json);
}

TEST(test_init_ok)
{
    setenv("RABBITMQ_HOST", "rabbit.local", 1);
    char *json = rabbit_init();
    ASSERT("200", json_status(json) == 200);
    free(json);
}

int main(void)
{
    RUN(test_init_missing);
    RUN(test_init_ok);
    printf("rabbit C tests passed\n");
    return 0;
}

File: queues/rabbit/Python/tests/test_rabbitclient.py
import sys
from pathlib import Path
from unittest.mock import MagicMock, patch

import pytest

sys.path.insert(0, str(Path(__file__).resolve().parent.parent))
import rabbitclient


@pytest.fixture(autouse=True)
def env(monkeypatch):
    monkeypatch.setenv("RABBITMQ_HOST", "rabbit.local")


def test_init_missing(monkeypatch):
    monkeypatch.delenv("RABBITMQ_HOST", raising=False)
    monkeypatch.delenv("RABBITMQ_URL", raising=False)
    assert rabbitclient.Init()["status_code"] == 601


def test_publish():
    mock_pika = MagicMock()
    conn = MagicMock()
    ch = MagicMock()
    mock_pika.BlockingConnection.return_value = conn
    mock_pika.URLParameters.return_value = object()
    conn.channel.return_value = ch
    with patch.dict(sys.modules, {"pika": mock_pika}):
        r = rabbitclient.Publish("q", {"n": 1})
    assert r == {"status_code": 200}
